File: backend/app/agent/tools.py
from dataclasses import dataclass


@dataclass(frozen=True)
class VocabularyEntry:
    word: str
    definition: str
    example: str


def lookup_word(word: str) -> VocabularyEntry:
    """Development dictionary tool; replace with a real provider later."""
    normalized = word.strip().lower()
    return VocabularyEntry(
        word=normalized,
        definition=f"A learner-friendly definition for '{normalized}'.",
        example=f"Try using '{normalized}' in a sentence about your day.",
    )


def create_exercise(topic: str, level: str) -> str:
    return f"Write three {level}-level sentences about {topic}."

File: backend/app/agent/tools.ts
/**
 * Agent tool contracts requested by the project structure.
 *
 * The FastAPI runtime uses tools.py. This TypeScript file documents equivalent
 * contracts for frontend/Node integrations without mixing TypeScript execution
 * into the Python service.
 */

export type VocabularyEntry = {
  word: string;
  definition: string;
  example: string;
};

export function lookupWord(word: string): VocabularyEntry {
  const normalized = word.trim().toLowerCase();
  return {
    word: normalized,
    definition: `A learner-friendly definition for "${normalized}".`,
    example: `Try using "${normalized}" in a sentence about your day.`,
  };
}

export function createExercise(topic: string, level: string): string {
  return `Write three ${level}-level sentences about ${topic}.`;
}
